src/api/children.js

```diff
@@ -7,19 +7,12 @@ define(function () {
       return node;
     }
 
-    var treeWalker = document.createTreeWalker(node, NodeFilter.SHOW_ALL, null, false);
-    var previousNode = treeWalker.currentNode;
-    if (treeWalker.firstChild()) {
-      // TODO: build list of non-empty elements (used elsewhere)
-      // Do not include non-empty elements
-      if (treeWalker.currentNode.nodeName === 'BR') {
-        return previousNode;
-      } else {
-        return firstDeepestChild(treeWalker.currentNode);
-      }
-    } else {
-      return treeWalker.currentNode;
+    var child = node.firstChild;
+    if( child.nodeName === 'BR' ) {
+      return node;
     }
+
+    return firstDeepestChild(child);
```

Comment:
I don't see how recursion is more efficient than the Treewalker solution I'm afraid.